src/Flight/FlightController.java

```diff
@@ -1,22 +1,24 @@
 package Flight;
 
+import java.time.LocalDateTime;
 import java.util.Scanner;
 
 public class FlightController {
     // Attributes
-    private Flight[] flights = new Flight[5];
+
     public Scanner scanner;
+    private final FlightService flightService;
 
     // Constructors - Dependency Injection - any dependent objects are provided at initiliazation
-    public FlightController(Scanner scanner) {
+    public FlightController(Scanner scanner, FlightService flightService) {
         this.scanner = scanner;
+        this.flightService = flightService;
```

Comment:
Is this trying to modify a Final object? I don't believe you can modify anything that has been declared final.